nifi-nar-bundles/nifi-update-attribute-bundle/nifi-update-attribute-processor/src/main/java/org/apache/nifi/processors/attributes/UpdateAttribute.java

```diff
@@ -166,7 +166,6 @@ public ValidationResult validate(String subject, String input, ValidationContext
             .description("If using state to set/reference variables then this value is used to set the initial value of the stateful variable. This will only be used in the @OnScheduled method " +
                     "when state does not contain a value for the variable.")
             .required(false)
-            .defaultValue("0")
             .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
```

Comment:
If we want to allow empty values, we shouldn't have the NON_EMPTY_VALIDATOR here. Also the description should mention that the user must Set Empty String if they want the value to exist but be empty. That doesn't change behavior per se, but not setting this property (versus setting it to empty) will affect the additional check for null property value below